pygmt/src/grd2cpt.py

```diff
@@ -0,0 +1,190 @@
+    :gmt-term:`COLOR_FOREGROUND`, and :gmt-term:`COLOR_NAN` from the
+    :gmt-docs:`gmt.conf <gmt.conf>` file or the command line will be used. This
+    default behavior can be overruled using the options ``background``,
+    ``overrule_bg`` or ``no_bg``.
+
+    The color model (RGB, HSV or CMYK) of the palette created by
+    :meth:`pygmt.grd2cpt` will be the same as specified in the header of the
+    master CPT. When there is no :gmt-term:`COLOR_MODEL` entry in the master
+    CPT, the :gmt-term:`COLOR_MODEL` specified in the
+    :gmt-docs:`gmt.conf <gmt.conf>` file or the ``color_model`` option will be
+    used.
+
+    Full option list at :gmt-docs:`grd2cpt.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+    transparency : int or float or str
+        Sets a constant level of transparency (0-100) for all color slices.
+        Append **+a** to also affect the fore-, back-, and nan-colors
+        [Default is no transparency, i.e., 0 (opaque)].
+    cmap : str
+        Selects the master color palette table (CPT) to use in the
+        interpolation. Full list of built-in color palette tables can be found
+        at :gmt-docs:`cookbook/cpts.html#built-in-color-palette-tables-cpt`.
+    background : bool or str
+        Select the back- and foreground colors to match the colors for lowest
+        and highest *z*-values in the output CPT [Default (``background=True``
+        or ``background='o'``) uses the colors specified in the master file, or
+        those defined by the parameters :gmt-term:`COLOR_BACKGROUND`,
+        :gmt-term:`COLOR_FOREGROUND`, and :gmt-term:`COLOR_NAN`]. Use
+        ``background='i'`` to match the colors for the lowest and highest
+        values in the input (instead of the output) CPT.
+    color_model :
+        [**R**\|\ **r**\|\ **h**\|\ **c**][**+c**\ [*label*]].
+        Force output CPT to be written with r/g/b codes, gray-scale values or
+        color name (**R**, default) or r/g/b codes only (**r**), or h-s-v codes
+        (**h**), or c/m/y/k codes (**c**).  Optionally or alternatively, append
+        **+c** to write discrete palettes in categorical format. If *label* is
+        appended then we create labels for each category to be used when the
+        CPT is plotted. The *label* may be a comma-separated list of category
+        names (you can skip a category by not giving a name), or give
+        *start*\[-], where we automatically build monotonically increasing
+        labels from *start* (a single letter or an integer). Append ``-`` to
+        build ranges *start*-*start+1* instead.
+    nlevels : bool or int or str
+        Set to ``True`` to create a linear color table by using the grid
+        z-range as the new  limits in the CPT. Alternatively, set *nlevels*
+        and to resample the color table into nlevels equidistant slices.
```

Comment:
```suggestion
        z-range as the new limits in the CPT. Alternatively, set *nlevels*
        to resample the color table into *nlevels* equidistant slices.
```

GMT 6.2.0 (unreleased) has some new options here (see https://github.com/GenericMappingTools/gmt/pull/3693, so we might need to update this in PyGMT v0.4.0.